src/components/shared/Slider/Rail.tsx

```diff
@@ -1,15 +1,533 @@
+};
+
+const getMarkValue = (step: number, markIndex: number): number => step * markIndex;
+const getMarkValues = (
+  step: number,
+  markPositions: number[],
+): number[] => markPositions.map((_, index) => getMarkValue(step, index));
+
+const createMarks = ({
+  positions,
+  step,
+  stepDistance,
+  style,
+  theme,
+  mark,
+  markWidth,
+  disabled,
+  onMarkPress,
+}: {
+  positions: number[];
+  step: number;
+  stepDistance: number;
+  style: ViewStyle;
+  theme: MarkThemeType;
+  mark?: React.ReactElement;
+  markWidth: number;
+  disabled?: boolean;
+  onMarkPress?: (value: number, position: number, index: number) => void | Promise<void>;
+}): React.ReactElement[] => {
+  const halfStepDistance = stepDistance / 2;
+  const fineTunedHalfStepDistance = halfStepDistance - (markWidth / 2);
+
+  return positions.map((position: number, index: number): React.ReactElement => {
+    const handlePress = (): void => {
+      if (disabled) {
+        return;
+      }
+
+      const value = getMarkValue(step, index);
+      if (onMarkPress) {
+        onMarkPress(value, position, index);
+      }
+    };
+
+    return (
+      <TouchableWithoutFeedback key={position} onPress={handlePress}>
+        <MarkPositioner width={stepDistance} position={position - fineTunedHalfStepDistance}>
+          {mark || <Mark style={style} theme={theme} />}
+        </MarkPositioner>
+      </TouchableWithoutFeedback>
+    );
+  });
+};
+
+const Rail: FC<Props> = ({
+  testID,
+  style = {},
+  markStyle = {},
+  mark,
+  customMarkWidth,
+  hideMark = false,
+  step = 1,
+  pixelsPerStep = 20,
+  markCount,
+  startMark = true,
+  endMark = true,
+  dark,
+  lightblue,
+  blue,
+  inverted = false,
+  disabled = false,
+  onInit,
+  onMarkPress,
+}) => {
+  const handleInit = (markValues: number[], markPositions: number[]): void => {
+    if (onInit) {
+      onInit(markValues, markPositions);
+    }
+  };
+
+  if (mark && !customMarkWidth) {
+    throw Error('Must have customMarkWidth prop if it uses a cutsom mark.');
+  }
+
+  const railStyleToApply = StyleSheet.flatten(style);
+  const markStyleToApply = StyleSheet.flatten(markStyle);
+
+  const defaultTheme = getDefaultTheme({
+    dark,
+    lightblue,
+    blue,
+    inverted,
+    disabled,
+  });
+
+  const railThemeToApply: RailThemeType = {
+    backgroundColor: railStyleToApply.backgroundColor || defaultTheme.backgroundColor,
+  };
+
+  const markThemeToApply: MarkThemeType = {
+    backgroundColor: markStyleToApply.backgroundColor || defaultTheme.markColor,
+  };
+
+  const railWidth = isNil(railStyleToApply.width) ? DEFAULT.width : railStyleToApply.width as string | number;
+  const railWidthInt = parseInt(railWidth.toString());
+
+  const markWidth = isNil(mark)
+    ? isNil(markStyleToApply.width) ? DEFAULT.MARK.width : markStyleToApply.width as string | number
+    : customMarkWidth as number;
+  const markWidthInt = parseInt(markWidth.toString());
```

Comment:
I'm not sure if we have the authority but if possible, I believe bumping our typescript version to 3.7 will make code like this cleaner by using nullish coalescing.

`const railWidth = railStyleToApply.width ?? DEFAULT.width;`
`const markWidth = isNil(mark) ? markStyleToApply.width ?? DEFAULT.MARK.width : customMarkWidth`

I'm eager to hear your opinion on this!

Reply:
Thank you for your detail!
I wanted to use `nullish coalescing` but I don't know how many side-effects by using `typescript 3.7+`).
`@dooboo-ui/native` uses `typescript 3.6.4` now.
We should make a PR with `typescript 3.7` to `master`, not in `feat/slider` because it's not only for [Slider].

I'll try to split the logic into a method to remove the triple operations for now.
Thanks! :D

Reply:
Sure! I checked the [breaking changes for ts 3.7](https://devblogs.microsoft.com/typescript/announcing-typescript-3-7/#breaking-Changes) and I believe migrating from 3.6 to 3.7 won't be that much of a deal but of course the devil lies in the details...

Would you as the team leader like to make the PR or should I make it on our behalf?

Reply:
I made two functions `getRailWidth()` and `getMarkWidth()` to split them and remove complex triple operations. :D

Reply:
@tony3918 
You can make a PR whenever you want.
I'm actually not a leader for this open source, I'm just a contributor as you. :)